### PR TITLE
PR to merge master branch into PG13 branch

### PR DESCRIPTION
**Pull request by jrgemignani**
**PLEASE DO NOT MERGE** until thoroughly reviewed. Additionally, this merge needs to be appropriately squashed before merging.

PR for moving the PG13 branch up to the current master branch.

It contains all of the latest work in the master branch, which is currently at PostgreSQL version 13. When this PR is merged PG13 will be the latest and represent PostgreSQL version 13 as the master moves to PostgreSQL version 14.

### CODE REVIEW

**Review by muhammadshoaib — APPROVED** [review]

**Review by dehowef — COMMENTED** [review]
I took a look, paying particular attention to the parser files with which I am most familiar with. The changes and overall patch look good to me, and the things I noticed were mostly minor optional corrections.

File: src/backend/nodes/cypher_readfuncs.c
[the remark this answers is not in the document]
Comment (jrgemignani) [reply src/backend/nodes/cypher_readfuncs.c:1]: That is the variable name used in the DEFINE statement for the READ_*_FIELD macros.

File: .dockerignore
[the remark this answers is not in the document]
Comment (jrgemignani) [reply .dockerignore:1]: Not really needed for these files.

File: drivers/golang/install.bat
[the remark this answers is not in the document]
Comment (jrgemignani) [reply drivers/golang/install.bat:1]: Not really needed for these files.

File: src/backend/executor/cypher_merge.c [hunk 1]
```diff
@@ -202,27 +197,25 @@ static bool check_path(cypher_merge_custom_scan_state *css,
              */
             if (slot->tts_isnull[node->tuple_position - 1])
             {
+
```
Comment (rafsun42) [inline src/backend/executor/cypher_merge.c:200]: Extra space

**Review by rafsun42 — APPROVED** [review]
Looks good to me.

**Pull request merged**